### PR TITLE
Refactor test_transforms.py into classes

### PR DESCRIPTION
**Pull request by AnirudhDagar**
This is a follow-up PR as discussed [here (#3982 comment)](https://github.com/pytorch/vision/pull/3982#issuecomment-855714826) after porting of unittest to pytest. See #3945

@NicolasHug I apologize for the unusually large diff in this single PR, but there is no logical change other than a few that I'll explicitly mark. Please let me know if I can help in getting this reviewed in any possible way.

### CODE REVIEW

File: test/test_transforms.py [hunk 1]
```diff
@@ -195,209 +197,486 @@ def test_accimage_crop(self):
...
-    [22, ], (27, ),
-])
-@pytest.mark.parametrize('max_size', (None, 37, 1000))
-def test_resize(height, width, osize, max_size):
-    img = Image.new("RGB", size=(width, height), color=127)
-
-    t = transforms.Resize(osize, max_size=max_size)
-    result = t(img)
-
-    msg = "{}, {} - {} - {}".format(height, width, osize, max_size)
```
Comment (AnirudhDagar) [inline test/test_transforms.py:345]: This has been dropped completely as mentioned in [#3952 comment](https://github.com/pytorch/vision/pull/3952#discussion_r645977808)

File: test/test_transforms.py [hunk 2]
```diff
@@ -195,209 +197,486 @@ def test_accimage_crop(self):
...
-    osize = osize[0] if isinstance(osize, (list, tuple)) else osize
-    # If size is an int, smaller edge of the image will be matched to this number.
-    # i.e, if height > width, then image will be rescaled to (size * height / width, size).
-    if height < width:
-        exp_w, exp_h = (int(osize * width / height), osize)  # (w, h)
-        if max_size is not None and max_size < exp_w:
-            exp_w, exp_h = max_size, int(max_size * exp_h / exp_w)
-        assert result.size == (exp_w, exp_h), msg
```
Comment (AnirudhDagar) [inline test/test_transforms.py:353]: removed msg.

File: test/test_transforms.py [hunk 3]
```diff
@@ -195,209 +197,486 @@ def test_accimage_crop(self):
...
-    elif width < height:
-        exp_w, exp_h = (osize, int(osize * height / width))  # (w, h)
-        if max_size is not None and max_size < exp_h:
-            exp_w, exp_h = int(max_size * exp_w / exp_h), max_size
-        assert result.size == (exp_w, exp_h), msg
```
Comment (AnirudhDagar) [inline test/test_transforms.py:358]: removed msg.

File: test/test_transforms.py [hunk 4]
```diff
@@ -195,209 +197,486 @@ def test_accimage_crop(self):
...
-    else:
-        exp_w, exp_h = (osize, osize)  # (w, h)
-        if max_size is not None and max_size < osize:
-            exp_w, exp_h = max_size, max_size
-        assert result.size == (exp_w, exp_h), msg
```
Comment (AnirudhDagar) [inline test/test_transforms.py:363]: removed msg.

File: test/test_transforms.py [hunk 5]
```diff
@@ -141,6 +142,7 @@ def test_int_to_int_consistency(self, input_dtype, output_dtype):
         assert actual_max == desired_max
 
 
+# TODO: CI is skipping these tests, fix it
```
Comment (AnirudhDagar) [inline test/test_transforms.py:145]: Adding a TODO note here.

File: test/test_transforms.py [hunk 6]
```diff
@@ -1668,445 +2070,5 @@ def test_color_jitter():
...
-
-    t = transforms.RandomAffine(10, translate=[0.5, 0.3], scale=[0.7, 1.3], shear=[-10, 10, 20, 40])
-    for _ in range(100):
-        angle, translations, scale, shear = t.get_params(t.degrees, t.translate, t.scale, t.shear,
-                                                         img_size=img.size)
-        assert -10 < angle < 10
-        assert -img.size[0] * 0.5 <= translations[0] <= img.size[0] * 0.5, ("{} vs {}"
-                                                                            .format(translations[0], img.size[0] * 0.5))
-        assert -img.size[1] * 0.5 <= translations[1] <= img.size[1] * 0.5, ("{} vs {}"
-                                                                            .format(translations[1], img.size[1] * 0.5))
```
Comment (AnirudhDagar) [inline test/test_transforms.py:2085]: The assert msg here has been dropped for simplicity. See [#3952 comment](https://github.com/pytorch/vision/pull/3952#discussion_r645977808)

**Review by AnirudhDagar — COMMENTED** [review]
Also just to make sure the tests themselves remain unchanged.

For test_transforms.py,
Before this PR : 886 tests passed 5 skipped.
After this PR : (EDIT) 886 tests passed 5 skipped.

**Comment by AnirudhDagar** [comment]
@NicolasHug Please take a look whenever you have some time. Thanks :))

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @AnirudhDagar 

I see the value in grouping the "to tensor" and the "to pil" tests into their respective classes.
For other cases though it's not as obvious. For example, it's not obvious whether `test_randomresized_params` should be in `TestResize`  or in `TestRandomX`.

Since moving tests around does come with the cost of adding an extra step in the git blame (which isn't negligible when it comes to maintenance), I'd suggest to only keep the `TestToTensor` and `TestToPil` classes for now.

Hopefully this will also reduce the size of the PR, making it easier to review. I'm happy to consider further changes in the future, but I'd suggest to leave that for other PRs.

**Comment by AnirudhDagar** [comment]
Thanks for the feedback @NicolasHug! I had a similar doubt as well while grouping a few tests. I also understand your very valid point on git blame and maintenance.

As suggested, I'll update the PR to include `TestToTensor` and `TestToPil` only for now.

**Comment by AnirudhDagar** [comment]
@NicolasHug I've made the changes, hopefully now the PR should be manageable for a review. Tests are all green as well! :)

**Review by NicolasHug — APPROVED** [review]
thanks @AnirudhDagar

**Comment by NicolasHug** [comment]
For the rest I think the only really obvious ones would be to group the `test_adjust` ones, but even then I don't see a strong incentive to do so, as they all test different transforms after all. So I feel like we can leave the rest as-is.

Thanks a ton for your work @AnirudhDagar !

**Comment by AnirudhDagar** [comment]
Thanks for your valuable feedback and detailed reviews @NicolasHug ! Let me know if there are any other issues where I can help.

Ps. I'll try to work on re-enabling the broken cpp tests (`mnasnet` specifically) over the weekend.

**Comment by NicolasHug** [comment]
If you're interested, https://github.com/pytorch/vision/issues/4063 is not taken yet ;)

**Pull request merged**